content/blog/2021-07-tonic-0-5.md

```diff
@@ -0,0 +1,174 @@
+
+[gRPC-Web] is a protocol that allows clients to connect to gRPC services over
+HTTP/1.1, instead of the usual HTTP/2. A common use-case for gRPC-Web is
+JavaScript clients running in the browser.
+
+The new crate [tonic-web] allows regular Tonic servers to accept gRPC-Web
+requests without the need of an external proxy.
+
+Enabling gRPC-Web support is as easy as:
+
+```rust
+use tonic::transport::Server;
+// code generated by tonic-build
+use hello_world::greeter_server::{GreeterServer, Greeter};
+
+struct MyGreeter;
+
+#[tonic::async_trait]
+impl Greeter for MyGreeter {
+    // ...
+}
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    let addr = "0.0.0.0:3000".parse().unwrap();
+
+    let greeter = GreeterServer::new(MyGreeter);
+
+    // enable grpc-web support for our `greeter` service
+    let service = tonic_web::enable(greeter);
+
+    Server::builder()
+        // by default tonic servers only accept http2 requests
+        // so we have to enable receiving http1 as well
+        .accept_http1(true)
+        .add_service(service)
+        .serve(addr)
+        .await?;
+
+    Ok(())
+}
+```
+
+For more details see the [tonic-web crate](https://crates.io/crates/tonic-web).
+
+# Compression
+
+Tonic can now transparently compress and decompression requests, responses, and
+streams.
+
+Enabling compression on a client is done like so:
+
+```rust
+let client = GreeterClient::new(channel)
+    // compress requests
+    .send_gzip()
+    /// accept compressed responses
+    .accept_gzip();
+```
+
+And like so on the server:
+
+```rust
+let service = GreeterServer::new(greeter)
+    // accept compressed requests
+    .accept_gzip()
+    // compress responses, if supported by the client
+    .send_gzip();
+```
+
+Note this requires enabling the `compression` feature on both tonic and
+[tonic-build]. See the [docs][compression-docs] for more details.
+
+# Improved Tower integration
+
+Tonic has always had support for extending clients and servers through [Tower]
+but in 0.5 the new [`Server::layer`] method makes it even easier. We can add
+tracing and authorization to a service by using middleware from [tower-http]:
+
+```rust
+use tonic::transport::Server;
+use tower_http::{
+    auth::RequireAuthorizationLayer,
+    trace::TraceLayer,
+};
+
+// The stack of middleware our service will be wrapped in
+let layer = tower::ServiceBuilder::new()
+    // High level tracing of requests and responses
+    .layer(TraceLayer::new_for_grpc())
+    // Authorize all requests using a token
+    .layer(RequireAuthorizationLayer::bearer("my-secret-token"))
+    // Convert our `ServiceBuilder` into a `tower::Layer`
+    .into_inner();
+
+Server::builder()
+    // Apply our middleware stack to the server
+    .layer(layer)
+    .add_service(GreeterServer::new(MyGreeter)
+    .serve(addr)
+    .await?;
+```
+
+# More flexible interceptors
+
+Interceptors are lightweight middleware that can, among other things, be used to
```

Comment:
maybe "Interceptors" should be a link

Reply:
it might be worth including something about how interceptors are different from Tower middleware? maybe something about how they are part of the gRPC spec (are they? or are they just an interface that many gRPC libraries provide by convention?) while Tower is an abstraction that works with gRPC or other protocols.

not a blocker, just a thought.